src/main/java/hudson/remoting/Launcher.java

```diff
@@ -269,6 +275,73 @@ public void setNoCertificateCheck(boolean ignored) throws NoSuchAlgorithmExcepti
             depends = "-workDir")
     public boolean failIfWorkDirIsMissing = WorkDirManager.DEFAULT_FAIL_IF_WORKDIR_IS_MISSING;
     
+    @Option(name = "-tunnel",
+            metaVar = "HOST:PORT",
+            usage = "Connect to the specified host and port, instead of connecting directly to Jenkins. "
+                    + "Useful when connection to Jenkins needs to be tunneled. Can be also HOST: or :PORT, "
+                    + "in which case the missing portion will be auto-configured like the default behavior.")
+    public String tunnel;
+
+    @Deprecated
+    @Option(name = "-headless", usage = "(deprecated; now always headless)")
+    public boolean headlessMode;
+
+    @Option(name = "-url", usage = "Specify the Jenkins root URLs to connect to.")
+    public List<URL> urls = new ArrayList<>();
+
+    @Option(name = "-webSocket",
+            usage = "Make a WebSocket connection to Jenkins rather than using the TCP port.",
+            depends = "-url",
+            forbids = {
+                "-direct",
+                "-tunnel",
+                "-credentials",
+                "-proxyCredentials",
+                "-cert",
+                "-noCertificateCheck",
+                "-noKeepAlive"
+            })
+    public boolean webSocket;
+
+    @Option(name = "-webSocketHeader",
+            usage = "Additional WebSocket header to set, eg for authenticating with reverse proxies. To specify multiple headers, call this flag multiple times, one with each header",
+            metaVar = "NAME=VALUE",
+            depends = "-webSocket")
+    public Map<String, String> webSocketHeaders;
+
+    /**
+     * Connect directly to the TCP port specified, skipping the HTTP(S) connection parameter download.
+     * @since 3.34
+     */
+    @Option(name = "-direct",
+            metaVar = "HOST:PORT",
+            aliases = "-directConnection",
+            depends = "-instanceIdentity",
+            forbids = {"-jnlpUrl", "-url", "-tunnel"},
+            usage = "Connect directly to this TCP agent port, skipping the HTTP(S) connection parameter download. For example, \"myjenkins:50000\".")
+    public String directConnection;
+
+    /**
+     * The controller's instance identity.
+     * @see <a href="https://wiki.jenkins.io/display/JENKINS/Instance+Identity">Instance Identity</a>
```

Comment:
May as well switch to a non-deprecated page while we are touching this:
```suggestion
     * @see <a href="https://plugins.jenkins.io/instance-identity/">Instance Identity</a>
```